### ISSUE TITLE
Figureout how to serialize formdata with 'application/x-www-form-urlencoded' and 'multipart/form-data'

### ISSUE THREAD

**Issue by amarzavery** [Azure/ms-rest-js#8]
- This is what the [request library](https://github.com/request/request#forms) does. They have separate fields `"form"` and `"formData"` that need to be used based on the `"Content-Type"` in the request header.
We have added a formDataFilter to the request pipeline in [ms-rest (node sdk)](https://github.com/Azure/azure-sdk-for-node/pull/2366/files) to solve this issue.
- We are depending on the `"fetch"` library which does not have good support for this.      
   - https://github.com/github/fetch/issues/263 and 
   - https://github.com/matthew-andrews/isomorphic-fetch/issues/30 
The above 2 issues should help us in finding the right solution.

**Comment by RikkiGibson** [comment]
@amarzavery is this still an issue?

**Comment by amarzavery** [comment]
In that case we will need to use some form-data package. Usually form-data is simple key/value pair kind of object. However the difficult part is calculating the boundary value and this value needs to be set in the header (I guess along with content-type value). Thus using a standard package in node.js that does that is beneficial for us.

**Comment by angurusw** [comment]
dfsdfdsfsdaf

**Comment by justinwilaby** [comment]
@amarzavery - 

Have you considered node-fetch with form-data? The isomorphic implementation can be either of 2 ways: 

## 1. Using Globals and the DefinePlugin for webpack
```js
/**
 * Node compatibility - This strategy hides
 * node-only dependencies from being bundled
 * in a browser based project even for dev builds.
 */
declare var __NODE__: boolean;
if (__NODE__) { // code is unreachable for __NODE__ == false and is stripped out
  (new Function(`
  if (!('FormData' in global)) {
    global.FormData = require('form-data');
  }
  if (!('fetch' in global)) {
    global.fetch = require('node-fetch');
  }`))();
}
```

## 2. Bring your own fetch and FormData
```js
HTTPClient.fetch = isNode() ? require('node-fetch') ? window.fetch
HTTPClient.FormData = isNode() ? require('form-data') ? window.FormData;
```

The first method does not require explicit configuration by the implementor but uses globals and could theoretically cause conflicts however unlikely they may be.

The second method requires that statics be configured with fetch and FormData by the implementor but allow more flexibility to provide their own fetch implementation on Node. Obviously, this is a no-op for most browser implementations.

Proxy support for Node users would then be provided *similar* to this regardless of the above implementation:

```js
const nodeFetch = require('node-fetch');
const HttpsProxyAgent = require('https-proxy-agent');

declare function fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;

export const fetch = function (...args: any[]) {
  const [urlOrRequest, requestInit = {}] = args;
  // No Proxy
  const url: string = typeof urlOrRequest === 'string' ? urlOrRequest : urlOrRequest.url;
  if (!process.env.HTTPS_PROXY || (process.env.NO_PROXY && url.includes(process.env.NO_PROXY))) {
    return nodeFetch(...args);
  }
  // URL is first param attach the proxy
  // to the RequestInit
  const agent = new HttpsProxyAgent(process.env.HTTPS_PROXY);
  if (typeof urlOrRequest === 'string') {
    requestInit.agent = agent;
  } else {
    urlOrRequest.agent = agent;
  }
  return nodeFetch(urlOrRequest, requestInit);
};
```

This would allow ms-rest-js to standardize on fetch, solve several issues that currently exist and to provide the smallest number of dependencies for *browser* users as possible.

**Comment by jeremymeng** [comment]
Closing as it's fixed by #342.
